src/include/units/bits/external/hacks.h

```diff
@@ -60,7 +60,7 @@ namespace std {
 template<class T>
 concept default_constructible = constructible_from<T>;
 
-#elif COMP_CLANG
+#elif COMP_CLANG && false
```

Comment:
You could replace these with
```suggestion
#elif COMP_CLANG && UNITS_LIBCXX
```
which you explicitly define to `1` when compiling with libc++ and otherwise set to `0` above, like `UNITS_DOWNCAST_MODE`. Seems that there's much fixing to be done to support libc++, too.